runtime/litmus/src/tests/xcm_parachain/mod.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright 2020-2022 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::setup::*;
+use crate::{
+	xcm_config::{CurrencyId, CurrencyIdMultiLocationConvert},
+	Origin,
+};
+use cumulus_primitives_core::{ParaId, PersistedValidationData};
+use cumulus_primitives_parachain_inherent::ParachainInherentData;
+
+use frame_support::{
+	assert_noop, assert_ok,
+	traits::{Currency, PalletInfoAccess},
+};
+use frame_system::RawOrigin;
+use polkadot_parachain::primitives::{AccountIdConversion, RelayChainBlockNumber, Sibling};
+use sp_runtime::{traits::Convert, AccountId32};
+use xcm::prelude::*;
+// use xcm::latest::prelude::*;
+use xcm_simulator::TestExt;
+
+pub mod relay_sproof_builder;
+
+fn para_a_account() -> AccountId32 {
+	ParaId::from(1).into_account()
+}
+
+fn para_b_account() -> AccountId32 {
+	ParaId::from(2).into_account()
+}
+
+fn _sibling_a_account() -> AccountId32 {
+	use sp_runtime::traits::AccountIdConversion;
+	Sibling::from(1).into_account()
+}
+
+fn _sibling_b_account() -> AccountId32 {
+	use sp_runtime::traits::AccountIdConversion;
+	Sibling::from(2).into_account()
+}
+
+#[test]
+fn test_xtokens_recognize_multilocation() {
```

Comment:
The network setup and simulation look good to me.

However, the integration test is still missing quite a bit IMHO :(
Current test only verifies that non-supported MultiLocation will be filtered out (`MultiLocationsFilter`), but this has nothing to do with `pallet-asset-manager`.

What we are looking forward to is the test for asset transfer between sibling parachains, e.g. LIT -> xLIT -> LIT: if we can handle it correctly after asset registration, including the fee handling of `units_per_second`

Reply:
Actaully, the integration setup of relay+para is not complete. Several bottom implementation of test is still for an automatic execution of XCM. I put this as TODO already.